warehouse/upload.go

```diff
@@ -1239,7 +1255,13 @@ func (job *UploadJobT) setUploadStatus(statusOpts UploadStatusOpts) (err error)
 		}
 
 		if config.GetBool("Reporting.enabled", types.DEFAULT_REPORTING_ENABLED) {
-			application.Features().Reporting.GetReportingInstance().Report([]*types.PUReportedMetric{&statusOpts.ReportingMetric}, txn)
+			metricsArr := make([]*types.PUReportedMetric, 0, len(statusOpts.ReportingMetrics))
+
+			for i := range statusOpts.ReportingMetrics {
+				metricsArr = append(metricsArr, &statusOpts.ReportingMetrics[i])
```

Comment:
You are appending to the array `metricsArr` which already has a default length specified when created. You will be unecessary holding blank values in it. Can you double check if length of the two arrays is similar ?

Reply:
Doesn't this format of make ensure 0 length and capacity=len(statusOpts.ReportingMetrics)? This should avoid any chance of blanks right?

Reply:
you are right, this becomes the cap.  Due to indentation, I was not looking at the initial `0`.